src/content/docs/codestream/start-here/codestream-new-relic.mdx

```diff
@@ -0,0 +1,224 @@
+
+**Step 1: Create a repository entity**
+
+To create a repository entity use the `referenceEntityCreateOrUpdateRepository` API and make sure to save the GUID that
+is produced.  The API take the following parameters:
+* `accountId` - the integer account id for the account you want to add the Repository to
+* `url` - example `https://github.com/newrelic/beta-docs-site.git`
+* `name` - example: `newrelic/beta-docs-site`
+```
+mutation {
+  referenceEntityCreateOrUpdateRepository(repositories: [{accountId: [YOUR_ACCOUNT_ID], name: "[REPO_NAME]", url: "[REPO_URL]"}]) {
+    created
+    failures {
+      guid
+      message
+      type
+    }
+  }
+}
+```
+In order to find back the entity you have created you can use a query like the following.  Note that the URL you
+provided to `referenceEntityCreateOrUpdateRepository` gets saved as an entity tag.
+```
+{
+  actor {
+    entitySearch(query: "name = 'a name' OR tags.url = 'a url'") {
+      count
+      query
+      results {
+        entities {
+          guid
+          name
+          tags {
+            key
+            values
+          }
+        }
+      }
+    }
+  }
+}
+
+```
+**Step 2: Associate the repository entity to your application entity**
+
+First find the GUID for the application you want to associate your repository too.  
+
+Parameters:
+* `sourceEntityGuid` - the entity GUID of the application
+* `targetEntityGuid` - the entity GUID of your Repository
+* `type` - always `BUILT_FROM`
+
+```
+mutation {
+  entityRelationshipUserDefinedCreateOrReplace(sourceEntityGuid: "", targetEntityGuid: "", type: BUILT_FROM) {
+    errors {
+      message
+      type
+    }
+  }
+}
+```
+
+To see all entities related to your Repository you can do a query like this:
+```
+{
+  actor {
+    entity(guid: "[YOUR_REPOSITORY_GUID]]") {
+      relatedEntities(filter: {direction: BOTH, relationshipTypes: {include: BUILT_FROM}}) {
+        results {
+          target {
+            entity {
+              name
+              guid
+              type
+            }
+          }
+          type
+        }
+      }
+      name
+      type
+      tags {
+        values
+        key
+      }
+    }
+  }
+}
+
+```
+
+**Step 3: Cleanup (if needed)**
+
+Delete a repository with the following query:
+```
+mutation DeleteRepository {
+  entityDelete(guids: "[ENTITY_GUID_HERE]]") {
+    deletedEntities
+    failures {
+      message
+      guid
+    }
+  }
+}
+
+```
+
+  </Collapser>
+
+</CollapserGroup>
+
+## Install APM agents with CodeStream [#install-agents]
+
+Requirements for installing New Relic APM agents via CodeStream:
+* [New Relic account](https://newrelic.com/signup)
+* [New Relic user key](https://one.newrelic.com/launcher/api-keys-ui.api-keys-launcher)
+* [CodeStream and New Relic connection](#connect)
+* A supported language application codebase
+    * .NET
```

Comment:
As of right now we are going to support .NET Core (support for .NET Framework may not be ready for GA)

Reply:
Good to know